Core/Src/BMP.c

```diff
@@ -178,24 +178,21 @@ static bool _readCalibration(void)
         return false;
     }
 
-    //TODO: REZOLVA WARNING
-    b1.AC1 = ((int16_t) calibValues[0] << 8) | calibValues[1];
-    b1.AC2 = ((int16_t) calibValues[2] << 8) | calibValues[3];
-    b1.AC3 = ((int16_t) calibValues[4] << 8) | calibValues[5];
-    b1.AC4 = ((uint16_t) calibValues[6] << 8) | calibValues[7];
-    b1.AC5 = ((uint16_t) calibValues[8] << 8) | calibValues[9];
-    b1.AC6 = ((uint16_t) calibValues[10] << 8) | calibValues[11];
-    b1.B1 = ((int16_t) calibValues[12] << 8) | calibValues[13];
-    b1.B2 = ((int16_t) calibValues[14] << 8) | calibValues[15];
-    b1.MB = ((int16_t) calibValues[16] << 8) | calibValues[17];
-    b1.MC = ((int16_t) calibValues[18] << 8) | calibValues[19];
-    b1.MD = ((int16_t) calibValues[20] << 8) | calibValues[21];
+    b1.AC1 = (int16_t)(calibValues[0] << 8 | calibValues[1]);
+    b1.AC2 = (int16_t)(calibValues[2] << 8 | calibValues[3]);
+    b1.AC3 = (int16_t)(calibValues[4] << 8 | calibValues[5]);
+    b1.AC4 = (uint16_t)(calibValues[6] << 8 | calibValues[7]);
+    b1.AC5 = (uint16_t)(calibValues[8] << 8 | calibValues[9]);
+    b1.AC6 = (uint16_t)(calibValues[10] << 8 | calibValues[11]);
+    b1.B1 = (int16_t)(calibValues[12] << 8 | calibValues[13]);
+    b1.B2 = (int16_t)(calibValues[14] << 8 | calibValues[15]);
+    b1.MB = (int16_t)(calibValues[16] << 8 | calibValues[17]);
+    b1.MC = (int16_t)(calibValues[18] << 8 | calibValues[19]);
+    b1.MD = (int16_t)(calibValues[20] << 8 | calibValues[21]);
```

Comment:
if b1 is a static variable use the naming convention you used before.

Also b1 is not a very clear name. Name it something more inspiring.